page_iterator.go

```diff
@@ -174,11 +186,11 @@ func (pI *PageIterator[T]) enumerate(callback func(item T) bool) bool {
 	for i := pI.pauseIndex; i < len(pageItems); i++ {
 		keepIterating = callback(pageItems[i])
 
+		// Set pauseIndex so that we know where to resume from.
```

Comment:
this fixes a bug where `Iterate` returns the final item indefinitely after all items have been enumerated.